samcli/commands/init/interactive_init_flow.py

```diff
@@ -149,57 +153,36 @@ def _generate_from_app_template(
         _package_schemas_code(runtime, schemas_api_caller, schema_template_details, output_dir, name, location)
 
 
-def _get_runtime(runtime):
-    if not runtime:
-        choices = list(map(str, range(1, len(INIT_RUNTIMES) + 1)))
-        choice_num = 1
-        click.echo("\nWhich runtime would you like to use?")
-        for r in INIT_RUNTIMES:
-            msg = "\t" + str(choice_num) + " - " + r
-            click.echo(msg)
-            choice_num = choice_num + 1
-        choice = click.prompt("Runtime", type=click.Choice(choices), show_choices=False)
-        runtime = INIT_RUNTIMES[int(choice) - 1]  # zero index
-    return runtime
+def _get_app_template_choice(templates):
+    chosen_template = templates[0]
+    if len(templates) > 1:
+        click.echo("\nSelect your starter template")
+        click_template_choices = []
+        for idx, template in enumerate(templates):
+            click.echo("\t{index} - {name}".format(index=idx + 1, name=template["displayName"]))
+            click_template_choices.append(str(idx + 1))
+        template_choice = click.prompt("Template", type=click.Choice(click_template_choices), show_choices=False)
+        chosen_template = templates[int(template_choice) - 1]
+    return chosen_template
+
+
+def _get_choice_from_options(msg, options):
+    options_list = list(options.keys())
+    click_choices = []
+    for idx, option in enumerate(options_list):
+        click.echo("\t{index} - {name}".format(index=idx + 1, name=option))
+        click_choices.append(str(idx + 1))
+    choice = click.prompt(msg, type=click.Choice(click_choices), show_choices=False)
+    choosen = options_list[int(choice) - 1]
+    return choosen
 
 
 def _get_runtime_from_image(image):
     """
     Get corresponding runtime from the base-image parameter
     """
-    if not image:
-        choices = list(map(str, range(1, len(LAMBDA_IMAGES_RUNTIMES) + 1)))
-        choice_num = 1
-        click.echo("\nWhich base image would you like to use?")
-        for r in LAMBDA_IMAGES_RUNTIMES:
-            msg = "\t" + str(choice_num) + " - " + r
-            click.echo(msg)
-            choice_num = choice_num + 1
-        choice = click.prompt("Base image", type=click.Choice(choices), show_choices=False)
-        image = LAMBDA_IMAGES_RUNTIMES[int(choice) - 1]  # zero index
-
     runtime = image[image.find("/") + 1 : image.find("-")]
-    return image, runtime
-
-
-def _get_dependency_manager(dependency_manager, runtime):
```

Comment:
Was this dead code? just trying to understand its removal.

Reply:
Yes are was not need because 

1. First the method was confusing, inspired good coding practice it should return only the runtime derived from the the image name. 
2. The list of images supported can be generated from the combination of use-cases and package-type selected by the customer. Therefore all this method `_get_runtime_from_image` needs to only get the runtime from the image name.
3. Currently `_get_dependency_manager` depends on `RUNTIME_TO_DEPENDENCY_MANAGERS` in the runtime.py file which is a mapping of runtimes and its dependencies. However in this design, this is not the case as the dependency manager can be easily gotten from the template choosen by the customer. This promotes, readability, ease of implementation and consistency